src/components/Editor/QuillEditor.jsx

```diff
@@ -0,0 +1,116 @@
+import React, { useEffect, useRef, useState } from "react";
+import "../../css/quillEditor.css";
+import { useFirebase } from "react-redux-firebase";
+import { useDispatch, useSelector } from "react-redux";
+import { Prompt } from "react-router-dom";
+import { setCurrentStep } from "../../store/actions";
+import * as Y from "yjs";
+import { QuillBinding } from "y-quill";
+import Quill from "quill";
+import QuillCursors from "quill-cursors";
+import { FirestoreProvider, getColor } from "@gmcfall/yjs-firestore-provider";
+
+import { onlineFirebaseApp } from "../../config";
+
+Quill.register("modules/cursors", QuillCursors);
+
+const QuillEditor = ({ id, data, tutorial_id }) => {
+  const [allSaved, setAllSaved] = useState(true);
+  const editorRef = useRef(null);
+  const containerRef = useRef(null);
+  let noteID = id || "test_note";
+  const firebase = useFirebase();
+  const dispatch = useDispatch();
+  const basePath = ["cl_codelabz", "organization", "codelabzorg", tutorial_id];
+  var provider, binding;
+  var ydoc;
+
+  const currentUserHandle = useSelector(
+    ({
+      firebase: {
+        profile: { handle }
+      }
+    }) => handle
+  );
+
+  useEffect(() => {
+    setAllSaved(true);
+  }, [id]);
+
+  useEffect(() => {
+    try {
+      if (!ydoc) {
+        ydoc = new Y.Doc();
+
+        // on updating text in editor this gets triggered
+        ydoc.on("update", () => {
```

Comment:
Thanks for adding comment, makes it easier to review @shivareddy6 👌

Reply:
Good to know its helping you out!